crates/kitsune_p2p/kitsune_p2p/src/gossip/simple_bloom.rs

```diff
@@ -0,0 +1,548 @@
+use crate::agent_store::AgentInfoSigned;
+use crate::types::gossip::*;
+use crate::types::*;
+use ghost_actor::dependencies::tracing;
+use kitsune_p2p_types::config::*;
+use kitsune_p2p_types::metrics::*;
+use kitsune_p2p_types::tx2::tx2_api::*;
+use kitsune_p2p_types::tx2::tx2_utils::*;
+use kitsune_p2p_types::*;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+/// max send buffer size (keep it under 16384 with a little room for overhead)
+/// (this is not a tuning_param because it must be coordinated
+/// with the constant in PoolBuf which cannot be set at runtime)
+const MAX_SEND_BUF_BYTES: usize = 16000;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub enum MetaOpKey {
+    /// data key type
+    Op(Arc<KitsuneOpHash>),
+
+    /// agent key type
+    Agent(Arc<KitsuneAgent>, u64),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub enum MetaOpData {
+    /// data chunk type
+    Op(Arc<KitsuneOpHash>, Vec<u8>),
+
+    /// agent chunk type
+    Agent(AgentInfoSigned),
+}
+
+impl MetaOpData {
+    fn byte_count(&self) -> usize {
+        match self {
+            MetaOpData::Op(_, d) => {
+                36 /* hash */ + d.len()
+            }
+            MetaOpData::Agent(a) => {
+                36 /* hash */ + 64 /* sig */ + a.as_agent_info_ref().len()
```

Comment:
Do we have consts for these byte counts anywhere? Wasn't there talk of using 39 bytes for some hashes in kitsune?

Reply:
I'm starting to feel like the 36->39 in kitsune isn't worth the effort... but yeah, I switched these to use the actual current size of the hashes/signature so it won't break if we do.